sdk/identity/identity/src/credentials/defaultAzureCredential.ts

```diff
@@ -11,149 +11,152 @@ import {
+  return new ManagedIdentityCredential(options);
 }
 
-export class DefaultAzureDeveloperCliCredential extends AzureDeveloperCliCredential {
-  constructor(options?: DefaultAzureCredentialOptions) {
-    super({
-      processTimeoutInMs: options?.processTimeoutInMs,
+function createDefaultWorkloadIdentityCredential(
+  options?: DefaultAzureCredentialOptions,
+): TokenCredential {
+  const managedIdentityClientId =
+    (options as DefaultAzureCredentialClientIdOptions)?.managedIdentityClientId ??
+    process.env.AZURE_CLIENT_ID;
+  const workloadIdentityClientId =
+    (options as DefaultAzureCredentialClientIdOptions)?.workloadIdentityClientId ??
+    managedIdentityClientId;
+  const workloadFile = process.env.AZURE_FEDERATED_TOKEN_FILE;
+  const tenantId = options?.tenantId ?? process.env.AZURE_TENANT_ID;
+  if (workloadFile && workloadIdentityClientId) {
+    const workloadIdentityCredentialOptions: WorkloadIdentityCredentialOptions = {
       ...options,
-    });
+      tenantId,
+      clientId: workloadIdentityClientId,
+      tokenFilePath: workloadFile,
+    };
+    return new WorkloadIdentityCredential(workloadIdentityCredentialOptions);
   }
-}
-export class DefaultAzureCliCredential extends AzureCliCredential {
-  constructor(options?: DefaultAzureCredentialOptions) {
-    super({
-      processTimeoutInMs: options?.processTimeoutInMs,
+  if (tenantId) {
+    const workloadIdentityClientTenantOptions: WorkloadIdentityCredentialOptions = {
       ...options,
-    });
+      tenantId,
+    };
+    return new WorkloadIdentityCredential(workloadIdentityClientTenantOptions);
   }
+
+  // This is a smell to me, and I believe can also be replaced with a UnavailableDefaultCredential
+  return new WorkloadIdentityCredential(options as WorkloadIdentityCredentialOptions);
 }
 
-export class DefaultAzurePowershellCredential extends AzurePowerShellCredential {
-  constructor(options?: DefaultAzureCredentialOptions) {
-    super({
-      processTimeoutInMs: options?.processTimeoutInMs,
-      ...options,
-    });
+function createDefaultAzureDeveloperCliCredential(
+  options: DefaultAzureCredentialOptions = {},
+): TokenCredential {
+  const processTimeoutInMs = options.processTimeoutInMs;
+  return new AzureDeveloperCliCredential({ processTimeoutInMs, ...options });
+}
+
+function createDefaultAzureCliCredential(
+  options: DefaultAzureCredentialOptions = {},
+): TokenCredential {
+  const processTimeoutInMs = options.processTimeoutInMs;
+  return new AzureCliCredential({ processTimeoutInMs, ...options });
+}
+
+function createDefaultAzurePowershellCredential(
+  options: DefaultAzureCredentialOptions = {},
+): TokenCredential {
+  const processTimeoutInMs = options.processTimeoutInMs;
+  return new AzurePowerShellCredential({ processTimeoutInMs, ...options });
+}
+
+function createEnvironmentCredential(options: DefaultAzureCredentialOptions = {}): TokenCredential {
```

Comment:
You can start to see the benefits here, and I kept it as small as possible just as an example, but if we are missing data we know the credential needs let's just not create the credential